### PR TITLE
update code to use standard Zen Cart functionallity

### PR DESCRIPTION
**Pull request by Zen4All**
[Review on Reviewable](https://reviewable.io/reviews/mc12345678/easypopulate-4.0/7)

### CODE REVIEW

File: admin/easypopulate_4.php [hunk 1]
```diff
@@ -213,20 +213,20 @@
 /* @var $epdlanguage_query type array */
 //$epdlanguage_query = $db->Execute("SELECT languages_id, name FROM ".TABLE_LANGUAGES." WHERE code = '".DEFAULT_LANGUAGE."'");
 if (!defined(DEFAULT_LANGUAGE)) {
-  $epdlanguage_query = ep_4_query("SELECT languages_id, code FROM " . TABLE_LANGUAGES . " ORDER BY languages_id LIMIT 1");
```
Comment (mc12345678) [inline admin/easypopulate_4.php]: This style of query offers in house error management and does not crash to resolve the issue by use of the ZC /cache or /logs directory or some other "outside" location, but instead provide a debug file that is readily available "on screen" for review.

File: admin/easypopulate_4.php [hunk 2]
```diff
@@ -213,20 +213,20 @@
...
+$epdlanguage_query = "SELECT languages_id, name FROM " . TABLE_LANGUAGES . " WHERE code = '" . DEFAULT_LANGUAGE . "'";
+$epdlanguage = $db->Execute($epdlanguage_query);
+if ($epdlanguage->RecordCount() > 0) {
+  $epdlanguage_id = $epdlanguage->fields['languages_id'];
+  $epdlanguage_name = $epdlanguage->fields['name'];
 } else {
   exit("EP4 FATAL ERROR: No default language set."); // this should never happen
 }
 
-$langcode = ep_4_get_languages(); // array of currently used language codes ( 1, 2, 3, ...)
```
Comment (mc12345678) [inline admin/easypopulate_4.php]: This style of query offers in house error management and does not crash to resolve the issue by use of the ZC /cache or /logs directory or some other "outside" location, but instead provide a debug file that is readily available on screen for review.  This and other similar potential replacements were considered in recent upgrade of the code.  That said the below manufacturers_query was not "refactored" to use this style of query last year in light of other issues trying to be addressed at the time the most of which was to ensure compatibility with ZC 1.5.4.

**Comment by Zen4All** [comment]
Ok, I understand your point of view, so I'll leave the rest of the code alone. Though in my opinion it over complicates the code :wink:

**Comment by mc12345678** [comment]
What we do for backwards compatibility... There WILL come a time when the "option" won't be available to use mysql_ type codes. I almost changed to ZC style queries when upgrading, but then I remembered the times I made an error in an import file and was thankful for the error logging introduced by chadd...

Yes there are ZC plugins that can support, but didn't want to take this "backwards" or force a different style of usage. Besides that, as a benefit of how the errors are handled, processing continues even if one "row" has an error in it rather than fatally stopping.

**Comment by mc12345678** [comment]
Also, don't get me wrong there may be other code improvements to be made and I appreciate the previous fix. Certainly would welcome a dutch or other translation if you are able. Particularly am interested in an italian translation seeing as the italian group won't allow me to have an account with. Them to request the same from there.

**Comment by Zen4All** [comment]
I'll see what I can do about the Dutch translation. For my Italian is as bad as my French and Spanish :laughing: . Maybe you could ask 100asa(https://www.zen-cart.com/member.php?31874-100asa) he is italian

**Pull request closed without merging**